web-console/future-openshift/contextual-help/contextual-help.md

```diff
@@ -0,0 +1,85 @@
+  * Rule of Thumb: Is it possible for a knowledgeable user to accomplish the thing being described without help? Or is information/data in the help necessary? If the latter, then the missing information belongs in the UI; it should not only exist in the help.
+* Use the Hint Pattern to elucidate obvious next steps.
+  * If there’s something a user should do (that we cannot do for them), use the Hint pattern to describe or provide a shortcut (link, button) for taking that next step.
+  * When the next step has been completed, users should not see the hint again for that resource.
+* Use the Infopop pattern to provide information for things users may have to do, but which may not be obvious.
+  * Use this pattern when we’re not sure when (or even if) a user may need the information.
+  * This pattern is useful when users may need to do something outside of the product UI to accomplish a goal.
+* Target help to intermediate users
+  * The majority of our users will be intermediate users.
+  * Being a beginner is not a permanent condition -- beginning users become intermediate users.
+  * Intermediate and advanced users will understand basic concepts and functions.
+* Onboarding and general information
+  * **This document does not specifically address onboarding or general information.**
+  * We should be looking at other ways to provide onboarding, or more general information to beginning users.
+  * This information should be available to novice users, but should be out of the way of intermediate and advanced users.
+
+## Hint Pattern
+The hint pattern is similar to an informational inline notification, but with additional emphasis (blue highlight). The intention is to use this for specific use cases where we can anticipate the next step a user should take, or to introduce new features. The hint pattern involves a one-step reminder, explanation, or call to action.
+
+### Potential Use cases
+1. To provide contextual help on using **new features**
+2. To provide hints on a flow/ the **next steps** a user should be thinking about.
+
+#### New Features
+![hint pattern - new features](img/hint-new-features.png)
+
+#### Next Steps
+![hint pattern - next steps](img/hint-next-step.png)
+
+### Design
+#### General
+![hint pattern](img/hint-general.png)
+* Hint area is shown in context of where it applies
+  * This may be above the page title for cases where the hints apply to the entire page of the application.
+  * This may be below a specific tab or area on a page for cases where the hints only apply to a portion.
+* Hint area is highlighted in blue
+* Hint must have a title (h3 style)
+* Hint must have some body text (body-copy style)
+* There can optionally be external links within the body text
+* There can optionally be a primary action below the body text
+* Title, body text, and primary action are left aligned
+* For cases without a primary action, there can optionally be an X in the top right corner to dismiss the message
```

Comment:
Are there cases where we would want to show a hint pattern with no primary action and no way to dismiss the message?

Also, wondering whether we'd want to specifically say that for cases with a primary action, do _not_ include an X. I'm wondering particularly about cases like "Create Service Binding", where we may want that message to be displayed until the action is taken.

Reply:
> Are there cases where we would want to show a hint pattern with no primary action and no way to dismiss the message?

I can't think of one. Do you have one in mind?
> 
> Also, wondering whether we'd want to specifically say that for cases with a primary action, do _not_ include an X. I'm wondering particularly about cases like "Create Service Binding", where we may want that message to be displayed until the action is taken.

In the service binding case, users *could* create that via the CLI, or by importing yaml. They don't have to use the UI; we are just suggesting what they should do and putting that action in front of them. Seems like if they choose to ignore or dismiss the message, then it is on them.

What I do think would be useful is to show the same message the next time they access the page if action hasn't been taken. So they'd only dismiss the message for that session.

Reply:
> I can't think of one. Do you have one in mind?

Me neither. Seems like "can optionally" on line 48 could be changed to "should" then.